### DISCUSSION TITLE
Unsure how to deploy a NextJs app using firebase when using Nx

### DISCUSSION THREAD

**Discussion by Crocsx** [nrwl/nx#20967] [General]
I create an app using [@nx/next][1]. So basically a `nextJs 13` app using `app routing` using `nx`.

Now, I created my project, worked inside my `apps/projectName/app` folder, and create a few `libs/libName/lib` etc... like a normal Nx repository.

I now arrived at a step where I would like to build and deploy a first draft of my work. I setup firebase

```
firebase experiments:enable webframeworks
firebase init hosting
```

And followed the step adding `hosting` and `firestore` (since I also use firestore) and accepting the default config once it detected my `nextJs` project. 

I ended up with 

```
{
  "firestore": {
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  },
  "hosting": {
    "source": ".",
    "ignore": ["firebase.json", "**/.*", "**/node_modules/**"],
    "frameworksBackend": {
      "region": "asia-east1"
    }
  }
}
```

Now, as the route of my built project is `dist/apps/projectName` I changed to 

```
{
  "firestore": {
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  },
  "hosting": {
    "source": "dist/apps/projectName",
    "ignore": ["firebase.json", "**/.*", "**/node_modules/**"],
    "frameworksBackend": {
      "region": "asia-east1"
    }
  }
}
```

The problem is, running deploy channel now trigger the following: 

`firebase hosting:channel:deploy --project projectName testChannel`

> Error: Could not find the next executable.

When I create a new clean project without Nx, it s pretty straightforward but with Nx, I am not sure what I am supposed to deploy even tho it looks similar structure as my build folder. I am unsure what is the correct configuration to allow deployment of a NextJs app inside a NX repository using firebase. 

  [1]: https://nx.dev/nx-api/next

**Comment by pvorona** [comment]
Do you see `package.json` in the build directory?

**Comment by Crocsx** [comment]
Yeah there is `package.json`.

I digged a bit deeper and look like that Firebase is trying to build static 
files (index.html etc) and for that it want to rebuild the project, but it 
do not support monorepo environment...

If I run a pre deploy script on firebase and `npm i` this error will disappear but firebase will complain that it can't find the sources files that are outside of the current repo and it fail to build the project (because they are contained in other lib inside the monorepo but outside the `dist` folder) 

I guess it's a firebase limitation and I need to work around it in some way by making having NX generate the static file instead, or scrap 
firebase for the hosting part.
